guides/theming-your-components.md

```diff
@@ -1,135 +1,224 @@
 
-Use the Sass `@include` keyword to include a component's theme mixin wherever you're already
-including Angular Material's built-in theme mixins. 
+  // Get the primary color palette from the color-config.
+  $primary-palette: map.get($color-config, 'primary');
 
-```scss
-// Import library functions for theme creation.
-@import '~@angular/material/theming';
-
-// Include non-theme styles for core.
-@include mat-core();
-
-// Define your application's custom theme.
-$primary: mat-palette($mat-indigo);
-$accent:  mat-palette($mat-pink, A200, A100, A400);
-$theme: mat-light-theme((
-  color: (
-    primary: $primary,
-    accent: $accent,
-  )
-));
+  .my-carousel-button {
+    // Read the 500 hue from the primary color palette.
+    color: mat.get-color-from-palette($primary-palette, 500);
+  }
+}
 
-// Include theme styles for Angular Material components.
-@include angular-material-theme($theme);
+@mixin typography($theme) {
+  // Get the typography config from the theme.
+  $typography-config: mat.get-typography-config($theme);
 
-// Include theme styles for your custom components.
-@include candy-carousel-theme($theme);
+  .my-carousel {
+    font-family: mat.font-family($typography-config);
+  }
+}
 ```
 
+### Step 3: Add a "theme" mixin
 
-## Note: using the `mat-color` function to extract colors from a palette
-
-You can consume the theming functions and Material Design color palettes from
-`@angular/material/theming`. The `mat-color` Sass function extracts a specific color from a palette.
-For example:
+For convenience, we can add a `theme` mixin that includes both color and typography.
+This theme mixin should only emit the styles for each color and typography, respectively, if they
+have a config specified.
 
 ```scss
-// Import theming functions
-@import '~@angular/material/theming';
+// _carousel-theme.scss
+
+@use 'sass:map';
+@use '~@angular/material' as mat;
 
-.candy-carousel {
-  // Get the default hue for a palette.
-  color: mat-color($primary);
+@mixin color($theme) {
+  // Get the color config from the theme.
+  $color-config: mat.get-color-config($theme);
   
-  // Get a specific hue for a palette. 
-  // See https://material.io/archive/guidelines/style/color.html#color-color-palette for hues.
-  background-color: mat-color($accent, 300);
+  // Get the primary color palette from the color-config.
+  $primary-palette: map.get($color-config, 'primary');
   
-  // Get a relative color for a hue ('lighter' or 'darker')
-  outline-color: mat-color($accent, lighter);
+  .my-carousel-button {
+    // Read the 500 hue from the primary color palette.
+    color: mat.get-color-from-palette($primary-palette, 500);
+  }
+}
+
+@mixin typography($theme) {
+  // Get the typography config from the theme.
+  $typography-config: mat.get-typography-config($theme);
+
+  .my-carousel {
+    font-family: mat.font-family($typography-config);
+  }
+}
+
+@mixin theme($theme) {
+  $color-config: mat.get-color-config($theme);
```

Comment:
You end up extracting the `color-config` and `typography-config` twice to check null in this structure, is that best practice?